sdk/scheduler/src/main/java/com/mesosphere/sdk/offer/evaluate/TLSEvaluationStage.java

```diff
@@ -0,0 +1,276 @@
+        } catch (Exception e) {
+            LOGGER.error("Failed to get certificate", e);
+            return EvaluationOutcome.fail(
+                    this, "Failed because of exception: %s", e);
+        }
+
+        Collection<Protos.Volume> volumes = getExecutorInfoSecretVolumes(podInfoBuilder);
+
+        // Share keys to the container
+        Optional<Protos.ExecutorInfo.Builder> executorBuilder = podInfoBuilder.getExecutorBuilder();
+        Protos.TaskInfo.Builder taskBuilder = podInfoBuilder.getTaskBuilder(taskName);
+        if (executorBuilder.isPresent()) {
+            executorBuilder.get()
+                    .getContainerBuilder()
+                    .setType(Protos.ContainerInfo.Type.MESOS)
+                    .addAllVolumes(volumes);
+            taskBuilder.setExecutor(executorBuilder.get());
+        }
+
+        return EvaluationOutcome.pass(
+                this, null, "TLS certificate created and added to the pod");
+
+    }
+
+    private Collection<Protos.Volume> getExecutorInfoSecretVolumes(PodInfoBuilder podInfoBuilder) {
+        String filenameInContainer = podInfoBuilder
+                .getPodInstance()
+                .getPod()
+                .getTransportEncryption()
+                .get()
+                .getName();
+
+        HashMap<String, String> tlsSecrets = new HashMap<>();
+        tlsSecrets.put(
+                getSecretPath(podInfoBuilder, "certificate-crt"),
+                String.format("%s.crt", filenameInContainer));
+
+        tlsSecrets.put(
+                getSecretPath(podInfoBuilder, "private-key"),
+                String.format("%s.key", filenameInContainer));
+
+        Collection<Protos.Volume> volumes = new ArrayList<>();
+
+        for (Map.Entry<String, String> entry : tlsSecrets.entrySet()) {
+
+            volumes.add(Protos.Volume.newBuilder()
+                    .setSource(Protos.Volume.Source.newBuilder()
+                            .setType(Protos.Volume.Source.Type.SECRET)
+                            .setSecret(getReferenceSecret(entry.getKey()))
+                            .build())
+                    .setContainerPath(entry.getValue())
+                    .setMode(Protos.Volume.Mode.RO)
+                    .build());
+
+        }
+
+        return volumes;
+    }
+
+    private static Protos.Secret getReferenceSecret(String secretPath) {
+        return Protos.Secret.newBuilder()
+                .setType(Protos.Secret.Type.REFERENCE)
+                .setReference(Protos.Secret.Reference.newBuilder().setName(secretPath))
+                .build();
+    }
+
+    private void storeSecrets(
+            PodInfoBuilder podInfoBuilder, String certPEM, String privateKeyPEM) throws IOException, SecretsException {
+
+        LOGGER.info(String.format("Creating new secret: %s", getSecretPath(podInfoBuilder, "certificate-pem")));
+
+        secretsClient.create(
+                getSecretPath(podInfoBuilder, "certificate-crt"),
```

Comment:
Also, there's duplicated `getSecretPath` calls for the same name. I suggest extracting those to variables.